flink-yarn/src/main/java/org/apache/flink/yarn/YarnFlinkResourceManager.java

```diff
@@ -438,6 +438,8 @@ private void containersAllocated(List<Container> containers) {
 			numPendingContainerRequests = Math.max(0, numPendingContainerRequests - 1);
 			LOG.info("Received new container: {} - Remaining pending container requests: {}",
 				container.getId(), numPendingContainerRequests);
+			resourceManagerClient.removeContainerRequest(new AMRMClient.ContainerRequest(
```

Comment:
Is this safe to create a new container request object ? Maybe we should call `getMatchingRequests` and remove one of the returned results.

Reply:
getMatchingRequests only filter on priority, resourceName, capability. However, ContainerRequest also contains racks, nodes information.